src/posix/platform/infra_if.cpp

```diff
@@ -280,23 +282,41 @@ bool InfraNetif::IsRunning(void) const { return (GetFlags() & IFF_RUNNING) && Ha
 
 uint32_t InfraNetif::GetFlags(void) const
 {
+    uint32_t     flags = 0;
     int          sock;
     struct ifreq ifReq;
 
     OT_ASSERT(mInfraIfIndex != 0);
 
     sock = SocketWithCloseExec(AF_INET6, SOCK_DGRAM, IPPROTO_IP, kSocketBlock);
-    VerifyOrDie(sock != -1, OT_EXIT_ERROR_ERRNO);
+    VerifyOrExit(sock != -1, otLogWarnPlat("Failed to create socket: %s", strerror(errno)));
 
     memset(&ifReq, 0, sizeof(ifReq));
     static_assert(sizeof(ifReq.ifr_name) >= sizeof(mInfraIfName), "mInfraIfName is not of appropriate size.");
     strcpy(ifReq.ifr_name, mInfraIfName);
 
-    VerifyOrDie(ioctl(sock, SIOCGIFFLAGS, &ifReq) != -1, OT_EXIT_ERROR_ERRNO);
+    VerifyOrExit(ioctl(sock, SIOCGIFFLAGS, &ifReq) != -1, otLogWarnPlat("Failed to call ioctl: %s", strerror(errno)));
 
     close(sock);
 
-    return static_cast<uint32_t>(ifReq.ifr_flags);
+    flags = static_cast<uint32_t>(ifReq.ifr_flags);
+
+exit:
+    return flags;
+}
+
+void InfraNetif::SetIcmp6Socket(int aFd)
+{
+    VerifyOrExit(mInfraIfIcmp6Socket != aFd);
+
+    if (mInfraIfIcmp6Socket != -1)
+    {
+        close(mInfraIfIcmp6Socket);
```

Comment:
If a resource is not auto-released, it's probably better that a resource is released by who allocated it. I think the socket FD will be owned by a FileDescriptor object in OtDaemonServer, so the fd shouldn't be freed here to avoid double-free error.

Reply:
I prefer to let OpenThread own the fd. 

Reasons:
- When running on platforms other than Android, the socket is created by OT and should be closed by OT. If we don't let OT own the socket we'll have the guard `close` calls by macros, which is less convenient.
- On Android, if the socket is owned by another process, the socket could be closed without notifying OT. This may also cause issues on OT.